### PR TITLE
F/static viz/tester

### PR DESCRIPTION
**Pull request by losrebellos**
Since it is quite hard to test and debug the render of the `static viz` components, I have created a small `tester` for it which has:
- chart `type` selector
- default configuration `options`
- (very) small options `editor`
- renders the chart accordingly

image

__Test__

So I could test: https://github.com/metabase/metabase/issues/20559
1. go to http://localhost:3000/_internal/static-viz
2. select `combo-chart`
3. paste in the options editor:
```json
{
  "settings": {
    "type": "timeseries",
    "x": {
      "type": "timeseries"
    },
    "y": {
      "type": "linear"
    },
    "labels": {
      "left": "Sum of price",
      "bottom": "Created at"
    }
  },
  "series": [
    {
      "name": "Doohickey",
      "color": "#88bf4d",
      "yAxisPosition": "left",
      "type": "line",
      "data": [
        ["2016-01-01", 632.1204747596681],
        ["2017-01-01", 854.1890645795561],
        ["2018-01-01", 496.41837341454766],
        ["2019-01-01", 203.13287766366682]
      ]
    },
    {
      "name": "Gadget",
      "color": "#f9d45c",
      "yAxisPosition": "left",
      "type": "line",
      "data": [
        ["2016-01-01", 679.8187899707065],
        ["2017-01-01", 1059.110673224628],
        ["2018-01-01", 844.5153990418078],
        ["2019-01-01", 435.7411381508008]
      ]
    },
    {
      "name": "Gizmo",
      "color": "#a989c5",
      "yAxisPosition": "left",
      "type": "line",
      "data": [
        ["2016-01-01", 529.7120290619603],
        ["2017-01-01", 1080.2154999244008],
        ["2018-01-01", 997.9395339531328],
        ["2019-01-01", 227.05089265934825]
      ]
    },
    {
      "name": "Widget",
      "color": "#f2a86f",
      "yAxisPosition": "left",
      "type": "line",
      "data": [
        ["2016-01-01", 987.3777276797102],
        ["2017-01-01", 1014.6824907503134],
        ["2018-01-01", 912.2167116544371],
        ["2019-01-01", 195.03825709546499]
      ]
    }
  ]
}
```

### CODE REVIEW

**Review by ranquild — COMMENTED** [review]
@losrebellos WDYT about creating a story (for storybook) instead of using this custom tester? It would allow you to have predefined chart types, and it would be also possible to paste chart configuration as json. Here is an example https://github.com/metabase/metabase/blob/3f08385d72aa27ce49e08e48292b870d3a2a9889/frontend/src/metabase/core/components/Button/Button.stories.tsx#L14. To run storybook locally, use `yarn storybook`.

**Comment by losrebellos** [comment]
> @losrebellos WDYT about creating a story (for storybook) instead of using this custom tester? It would allow you to have predefined chart types, and it would be also possible to paste chart configuration as json. Here is an example
> 
> https://github.com/metabase/metabase/blob/3f08385d72aa27ce49e08e48292b870d3a2a9889/frontend/src/metabase/core/components/Button/Button.stories.tsx#L14
> 
> . To run storybook locally, use `yarn storybook`.

Sounds like a very good idea.
Will need to investigate how `storybook` will handle the JSON options for chart components tho.

**Comment by ranquild** [comment]
@losrebellos a better example https://61d4abcd959e8c003aa51d4c-yemzjcvtiw.chromatic.com/?path=/story/status-databasestatuslarge--aborted https://github.com/metabase/metabase/blob/42ff35d5d881b6e65e1cccb133fd787085fb7802/frontend/src/metabase/status/components/DatabaseStatusLarge/DatabaseStatusLarge.stories.tsx#L16 . See it's possible to set `databases` as json in story

File: frontend/src/metabase/static-viz/containers/StaticChart/types.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { STATIC_CHART_TYPES } from "./constants";
+
+type StaticChartType = typeof STATIC_CHART_TYPES[number];
+
+export interface IStaticChartProps {
```
Comment (ranquild) [inline frontend/src/metabase/static-viz/containers/StaticChart/types.ts:5]: We don't use `I` prefix in the core product

File: frontend/src/metabase/static-viz/containers/StaticChart/types.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+  type: StaticChartType;
+  options: any;
```
Comment (ranquild) [inline frontend/src/metabase/static-viz/containers/StaticChart/types.ts:7]: `unknown`?

**Review by ranquild — APPROVED** [review]
Looks good 🚀

File: frontend/src/metabase/static-viz/containers/StaticChart/types.ts (hunk 2 above)
Comment (losrebellos) [reply frontend/src/metabase/static-viz/containers/StaticChart/types.ts:7]: `unknown` is throwing a TypeScript error regarding passed props being incompatible...
IMHO, if `StaticChart` is allowing to render all chart components, all the components `props` should use the same interface.

**Pull request merged**